inc/components/class-image.php

```diff
@@ -192,6 +192,11 @@ public function set_config_for_size( string $image_size, $picture = false ) {
 			$size_config = $sizes[ $image_size ];
 			$this->set_config( 'image_size', $image_size );
 			$this->set_config( 'sources', $size_config['sources'] );
+
+			// You can set certain config values on a per-size basis instead of per-component
+			$this->set_config( 'aspect_ratio', $size_config['aspect_ratio'] ?? $this->config['aspect_ratio'] );
+			$this->set_config( 'retina', $size_config['retina'] ?? $this->config['retina'] );
+			$this->set_config( 'lazyload', $size_config['lazyload'] ?? $this->config['lazyload'] );
```

Comment:
Let's error proof this a bit more,
`$this->set_config( 'aspect_ratio', $size_config['aspect_ratio'] ?? $this->config['aspect_ratio'] ?? '' );`

That last `??`, just incase `$this->config['aspect_ratio']` is null too, we won't get a fatal error.